File: api/src/main/java/com/example/athletehub/dto/PublicProfileResponse.java
package com.example.athletehub.dto;

/**
 * Aggregate payload returned by `GET /api/users/{handle}`. The viewer-scoped
 * {@code iFollow} flag lets the client render the follow/unfollow button
 * without a second round-trip.
 */
public record PublicProfileResponse(
        PublicUserDto user,
        Integer followers,
        Integer following,
        Boolean iFollow
) {}

File: client/lib/models/responses/public_profile_response.dart
import '../public_user.dart';

/// Aggregate response from `GET /api/users/{handle}` — header + counters +
/// the viewer-scoped follow flag.
class PublicProfileResponse {
  PublicProfileResponse({
    required this.user,
    required this.followers,
    required this.following,
    required this.iFollow,
  });

  final PublicUser user;
  final int followers;
  final int following;
  final bool iFollow;

  factory PublicProfileResponse.fromJson(Map<String, dynamic> json) {
    return PublicProfileResponse(
      user: PublicUser.fromJson(json['user'] as Map<String, dynamic>),
      followers: (json['followers'] as num?)?.toInt() ?? 0,
      following: (json['following'] as num?)?.toInt() ?? 0,
      iFollow: json['iFollow'] as bool? ?? false,
    );
  }
}
